### PR TITLE
add {Alg, S} type parameters to AbstractODEIntegrator

### PR DESCRIPTION
**Pull request by Datseris**

### CODE REVIEW

File: src/common_interface/integrator_types.jl [hunk 1]
```diff
@@ -13,9 +13,10 @@ mutable struct DEOptions{SType,TstopType,CType}
     stop_at_next_tstop::Bool
 end
 
-abstract type AbstractSundialsIntegrator <: DiffEqBase.AbstractODEIntegrator end
+struct SundialsAlgorithm end
+abstract type AbstractSundialsIntegrator{S} <: DiffEqBase.AbstractODEIntegrator{SundialsAlgorithm, S} end
```
Comment (ChrisRackauckas) [inline src/common_interface/integrator_types.jl]: use the actual alg

File: src/common_interface/integrator_types.jl [hunk 2]
```diff
@@ -13,9 +13,10 @@ mutable struct DEOptions{SType,TstopType,CType}
     stop_at_next_tstop::Bool
 end
 
-abstract type AbstractSundialsIntegrator <: DiffEqBase.AbstractODEIntegrator end
+struct SundialsAlgorithm end
```
Comment (ChrisRackauckas) [inline src/common_interface/integrator_types.jl]: why?

Comment (Datseris) [reply src/common_interface/integrator_types.jl]: I think I got confused. What is their type? Tell me and I can use it. I commented in a commit:

> I don't know what "use the actual alg" means. Which is the "actual" one?  `CVODE_Adams`? `CVODE_BDF`?

Comment (ChrisRackauckas) [reply src/common_interface/integrator_types.jl]: Exactly. CVODE_Adams, CVODE_BDF, ARKODE, and IDA are the 4 algorithms here.

Comment (Datseris) [reply src/common_interface/integrator_types.jl]: `CVODEIntegrator` is one struct that has two algorithms. Should I just leave the type paramaeterized thenn?

Comment (ChrisRackauckas) [reply src/common_interface/integrator_types.jl]: nah, it needs to pass the algorithm through and add it to the integrator.

Comment (Datseris) [reply src/common_interface/integrator_types.jl]: I am sorry Chris, I don't know how this works. I'll kindly ask you to add the finishing touches here, while I move on to making PRs to the other repos listed in the DiffEq issue.

**Pull request closed without merging**